src/services/position_sync.py

```
@@ -0,0 +1,120 @@
+    client_id: int,
+    connect_timeout_seconds: float = 20.0,
+) -> int:
+    ib = IB()
+    try:
+        try:
+            ib.connect(host, port, clientId=client_id, timeout=connect_timeout_seconds)
+        except TimeoutError as exc:
+            raise RuntimeError(
+                "Timed out connecting to TWS/Gateway while fetching positions "
+                f"(host={host}, port={port}, client_id={client_id}, timeout={connect_timeout_seconds}s)."
+            ) from exc
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)